src/backend/distributed/commands/function.c

```diff
@@ -1328,6 +1328,12 @@ CreateFunctionStmtObjectAddress(Node *node, bool missing_ok)
 	FunctionParameter *funcParam = NULL;
 	foreach_ptr(funcParam, stmt->parameters)
 	{
+		if (funcParam->mode == FUNC_PARAM_TABLE || funcParam->mode == FUNC_PARAM_OUT)
```

Comment:
what about `FUNC_PARAM_INOUT` or `FUNC_PARAM_VARIADIC`? I think the latter is just input, but do we require to filter out `FUNC_PARAM_INOUT` as well?

 
I also find it confusing/dangerous to have multiple code-blocks in different files that calculates the function arguments. 

But I guess we cannot use `ObjectWithArgsFromOid()` here because it is not guaranteed that the input function has been created. Is my understanding accurate? 

Can we at least have an API, like `FuncParamTypeForObjectAddress` or such? And, use that function in both places? (and, any other places if needed)

Reply:
1) I think we shouldn't filter neither. 
I think that way because in plain PG, we can, say, drop a function with OUT params by excluding OUT params in the DROP FUNCTION statement; PG errors out if we don't exclude OUT params.
This is not the case for INOUT and VARIADIC params. PG requires these param modes in the function signature. So I think we should only filter OUT and TABLE params.

2) For these two relevant places, we don't have the same type. I mean, even if we had a function to filter args, what would be the parameter type of that function? Here we filter out the given arguments in a `CreateFunctionStatement`. But at the other place we do filtering, we fetch an existing function from PG, using the type `ObjectWithArgs`

Reply:
> I mean, even if we had a function to filter args, what would be the parameter type of that function?

It should just get FunctionParameterMode:

```C
bool
ShouldAddFunctionSignature(FunctionParameterMode mode)
{
   switch mode
     .... 
      return true:
   
     default:
       return false
}

Reply:
that makes sense, added, thank you